axum/src/routing/path_router.rs

```diff
@@ -36,6 +36,69 @@ where
     }
 }
 
+// Validates a path in compile time, used with the vpath macro.
+// Allow dead code needed because this is only used within the vpath macro
+// and if there's no call to it, then, there's no users of it.
+#[allow(dead_code)]
+#[doc(hidden)]
+pub const fn __private_validate_static_path(path: &'static str) -> &'static str {
+    if path.is_empty() {
+        panic!("Paths must start with a `/`. Use \"/\" for root routes")
+    }
+    if path.as_bytes()[0] != b'/' {
+        panic!("Paths must start with /");
+    }
+    path
+}
+
+#[rustversion::since(1.80)]
+#[macro_export]
+/// This macro checks that your string is the correct format to be used as a route,
+/// and fails in compilation time instead of runtime, if it's not.
+///
+/// This example will stop the compialtion:
+///
+/// ```compile_fail
+/// use axum::routing::*;
+/// use axum::vpath;
+///
+/// let handler = async {};
```

Comment:
Not sure what this is here for.

Reply:
ah, it's a stray line that I forgot to remove. sorry.